packages/vhd-cli/src/commands/compare.js

```diff
@@ -0,0 +1,111 @@
+import { VhdFile, VhdDirectory } from 'vhd-lib'
+
+import { getSyncedHandler } from '@xen-orchestra/fs'
+import { resolve } from 'path'
+import Disposable from 'promise-toolbox/Disposable'
+
+const compareObj = (src, dest, path) => {
+  for (const key in src) {
+    if (!Object.prototype.hasOwnProperty.call(src, key)) {
+      continue
+    }
+    let same = false
+    if (typeof src[key] !== typeof dest[key]) {
+      throw new Error(
+        `Error checking header : key ${path + '/' + key} is of type *${src[key]}* in source and *${dest[key]}* in dest`
+      )
+    }
+
+    if (typeof src[key] === 'object') {
+      compareObj(src[key], dest[key], path + '/' + key)
+      continue
```

Comment:
IMHO, `continue`, `break` and `returns` complexify the workflow and can lead to maintenance issues, they should be used only when providing clear advantages, otherwise prefer `if`/`else`.